lib/module-resolvers/file-resolver.js

```diff
@@ -24,6 +27,22 @@ class FileResolver {
 
         this._additional_search_paths = options.search_paths || search_paths();
 
+        // Also add the module root for all adapters.
+        this._adapter_roots = [];
+
+        for(let adapter of adapters.list()) {
+            // Skip built-in adapters or adapters for which the
+            // install path is not absolute (i.e. not actually available)
+            if (adapter.module !== '(builtin)' &&
+                path.isAbsolute(adapter.path)) {
```

Comment:
I think it would be better to include a flag indicating if the adapter was actually loaded (e.g. `adapter.loaded`) and use it here. Checking the path is definitely not a clean check adapter’s state.